### PR TITLE
Stop using native java rules as these are now deprecated usages. 

### LINKED ISSUE #247: Flag --incompatible_load_java_rules_from_bzl will break rules_kotlin in Bazel 1.2.1
Incompatible flag --incompatible_load_java_rules_from_bzl will break rules_kotlin once Bazel 1.2.1 is released.

Please see the following CI builds for more information:

* [:ubuntu: 18.04 (OpenJDK 11)](<a href="https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds/342#bf3c3933-3c67-46d7-9e4f-b65750150404" target="_blank">:ubuntu: 18.04 (OpenJDK 11)</a>)
* [:ubuntu: 16.04 (OpenJDK 8)](<a href="https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds/342#4203d46d-a180-4cd7-a83b-e4f626eda697" target="_blank">:ubuntu: 16.04 (OpenJDK 8)</a>)
* [:ubuntu: 18.04 (OpenJDK 11)](<a href="https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds/342#262d9734-0ed4-4c01-ae10-d2b5a7dd73b7" target="_blank">:ubuntu: 18.04 (OpenJDK 11)</a>)
* [:darwin: (OpenJDK 8)](<a href="https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds/342#77dad464-8587-4a5d-b0a4-faea39699152" target="_blank">:darwin: (OpenJDK 8)</a>)
* [:ubuntu: 18.04 (OpenJDK 11)](<a href="https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds/342#f24555d4-a050-4668-b87f-a6f46a8464b6" target="_blank">:ubuntu: 18.04 (OpenJDK 11)</a>)
* [:ubuntu: 18.04 (OpenJDK 11)](<a href="https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds/342#c340f9bd-c19b-4a69-b867-3fba6f7310f5" target="_blank">:ubuntu: 18.04 (OpenJDK 11)</a>)
* [RBE (:ubuntu: 16.04, OpenJDK 8)](<a href="https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds/342#e0effe17-f7f9-4cf4-ade0-04ea0cf2cfab" target="_blank">RBE (:ubuntu: 16.04, OpenJDK 8)</a>)

Questions? Please file an issue in https://github.com/bazelbuild/continuous-integration

**Important**: Please do NOT modify the issue title since that might break our tools.

### LINKED ISSUE #254: Flag --incompatible_restrict_string_escapes will break rules_kotlin in Bazel 1.2.1
Incompatible flag --incompatible_restrict_string_escapes will break rules_kotlin once Bazel 1.2.1 is released.

Please see the following CI builds for more information:

* [:ubuntu: 18.04 (OpenJDK 11)](<a href="https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds/342#bf3c3933-3c67-46d7-9e4f-b65750150404" target="_blank">:ubuntu: 18.04 (OpenJDK 11)</a>)
* [:ubuntu: 16.04 (OpenJDK 8)](<a href="https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds/342#4203d46d-a180-4cd7-a83b-e4f626eda697" target="_blank">:ubuntu: 16.04 (OpenJDK 8)</a>)
* [:darwin: (OpenJDK 8)](<a href="https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds/342#77dad464-8587-4a5d-b0a4-faea39699152" target="_blank">:darwin: (OpenJDK 8)</a>)
* [:ubuntu: 18.04 (OpenJDK 11)](<a href="https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds/342#f24555d4-a050-4668-b87f-a6f46a8464b6" target="_blank">:ubuntu: 18.04 (OpenJDK 11)</a>)
* [RBE (:ubuntu: 16.04, OpenJDK 8)](<a href="https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds/342#e0effe17-f7f9-4cf4-ade0-04ea0cf2cfab" target="_blank">RBE (:ubuntu: 16.04, OpenJDK 8)</a>)

Questions? Please file an issue in https://github.com/bazelbuild/continuous-integration

**Important**: Please do NOT modify the issue title since that might break our tools.

### PR DESCRIPTION
**Pull request by cgruber**
Fixes #247

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -65,11 +65,18 @@ rbe_autoconfig(
 )
 
 load("//kotlin:kotlin.bzl", "kotlin_repositories", "kt_register_toolchains")
-
 kotlin_repositories()
-
 kt_register_toolchains()
 
+http_archive(
```
Comment (jin) [inline WORKSPACE]: There's a vendored @rules_java in every Bazel installation. Do we need this here?

Comment (cgruber) [reply WORKSPACE]: Oh? Nice! I guess we do not. Though it semi-forces the kotlin rules to use whatever is in the current bazel... that definitely ties the lifecycles, but in practice, they're already tied, so it's no worse, and there's less hassle. I'll remove. Do I need to declare it or is it just available as `@rules_java`?

Comment (jin) [reply WORKSPACE]: It's available as @rules_java, no need to declare. Bazel prefixes and suffixes every WORKSPACE with some implicit repositories, and @rules_java is one of them.

Comment (cgruber) [reply WORKSPACE]: Nice - is there a list? (Also... done)

Comment (jin) [reply WORKSPACE]: here, https://source.bazel.build/bazel/+/master:src/main/java/com/google/devtools/build/lib/bazel/rules/java/jdk.WORKSPACE;l=3;drc=a6cd408876dd55c988a123af8237807bee549a5b has some but not all. 

There's no canonical list AFAIK. you can also get a hint by looking at Bazel's own WORKSPACE file here:

https://source.bazel.build/bazel/+/master:WORKSPACE;l=169-233;drc=168d0a7aa2af7c99a7ed2cd98b89fceb73aebd48

**Pull request merged**